linera-service/tests/linera_net_tests.rs

```diff
@@ -1408,6 +1408,142 @@ async fn test_wasm_end_to_end_counter(config: impl LineraNetConfig) -> Result<()
+    let constructor_argument = ConstructorArgs { the_supply };
+    let constructor_argument = constructor_argument.abi_encode();
+
+    let instantiation_argument = U256::abi_encode(&the_supply);
+
+    let (evm_contract, _dir) = get_evm_contract_path("tests/fixtures/erc20_shared.sol")?;
+
+    let application_id = client1
+        .publish_and_create::<EvmAbi, Vec<u8>, Vec<u8>>(
+            evm_contract.clone(),
+            evm_contract,
+            VmRuntime::Evm,
+            &constructor_argument,
+            &instantiation_argument,
+            &[],
+            None,
+        )
+        .await?;
+
+    let port1 = get_node_port().await;
+    let port2 = get_node_port().await;
+    let mut node_service1 = client1.run_node_service(port1, ProcessInbox::Skip).await?;
+    let mut node_service2 = client2.run_node_service(port2, ProcessInbox::Skip).await?;
+
+    let application1 = node_service1
+        .make_application(&chain1, &application_id)
+        .await?;
+
+    let application2 = node_service2
+        .make_application(&chain2, &application_id)
+        .await?;
+
+    // Checking the total supply
+
+    let total_supply = totalSupplyCall {};
+    let query = total_supply.abi_encode();
+    let query = EvmQuery::Query(query);
+    let result = application1.run_json_query(query).await?;
+    assert_eq!(read_evm_u256_entry(result), the_supply);
+
+    // Transferring to another user and checking the balances.
+
+    let mutation = transferCall {
+        to: address2,
+        value: transfer1,
+    };
+    let mutation = EvmQuery::Mutation(mutation.abi_encode());
+    application1.run_json_query(mutation).await?;
+
+    let query = balanceOfCall { account: address1 };
+    let query = EvmQuery::Query(query.abi_encode());
+    let result = application1.run_json_query(query).await?;
+    assert_eq!(read_evm_u256_entry(result), the_supply - transfer1);
+
+    let query = balanceOfCall { account: address2 };
+    let query = EvmQuery::Query(query.abi_encode());
+    let result = application1.run_json_query(query).await?;
+    assert_eq!(read_evm_u256_entry(result), transfer1);
+
+    // Transferring to another chain and checking the balances.
+
+    let chain_id: [u64; 4] = <[u64; 4]>::from(chain2.0);
+    let chain_id: [u8; 32] = linera_base::crypto::u64_array_to_be_bytes(chain_id);
+    let chain_id: B256 = chain_id.into();
+    let mutation = transferToChainCall {
+        chain_id,
+        value: transfer2,
+    };
+    let mutation = EvmQuery::Mutation(mutation.abi_encode());
+    application1.run_json_query(mutation).await?;
+
+    node_service2.process_inbox(&chain2).await?;
+
+    // Checking the balances on both chains.
+
+    let query = balanceOfCall { account: address1 };
+    let query = EvmQuery::Query(query.abi_encode());
+    let result = application1.run_json_query(query.clone()).await?;
+    assert_eq!(
+        read_evm_u256_entry(result),
+        the_supply - transfer1 - transfer2
+    );
```

Comment:
Can you show how we can transfer the tokens sent to a (different) `chain` as recipient?